src/main/docs/guide/generators.adoc

```diff
@@ -0,0 +1,25 @@
+To write a source generator you have a few options. If you only wish to support Java you can use the https://github.com/square/javapoet[JavaPoet API] that is forked into the `io.micronaut.sourcegen.javapoet` package. This fork includes support for Java 17+ constructors like Records.
```

Comment:
Can you add a note that JavaPoet is no longer being developed? That's why it's forked